src/sql/workbench/services/objectExplorer/browser/objectExplorerService.ts

```diff
@@ -246,7 +249,11 @@ export class ObjectExplorerService implements IObjectExplorerService {
 				nodeStatus.expandEmitter.fire(expandResponse);
 			}
 		}
+
 		if (!foundSession) {
+			// Just fire reponse with empty nodes for example: request from standalone SQL instance
+			// TODO Need to investigate why we don't get nodes only 'sometimes'
+			this._onExpandCompleteWithoutNodes.fire(expandResponse);
```

Comment:
You mentioned to me that you were seeing the session exist but not have any nodes in the map (which meant foundSession stayed as false). 

It would be good to first add an error log message for that then so it's clear when that happens. 

And then did you dig into why we weren't getting the node added to the map? Was it never coming in at all or was it coming in but just out of order from this callback being called?

Reply:
> You mentioned to me that you were seeing the session exist but not have any nodes in the map (which meant foundSession stayed as false).

> And then did you dig into why we weren't getting the node added to the map? Was it never coming in at all or was it coming in but just out of order from this callback being called?

> when you say "not having nodes" it's not that the node returned no nodes (empty list) but rather that the when the onNodeExpanded callback was called the internal map of nodes we keep for some reason doesn't have an entry for the node that was originally expanded, right?

Yes, not having any nodes, eventually meant the `expandResponse` was not being populated to the tree. The list of nodes is there, but it was not fed to the tree. Now it's done here: 
https://github.com/microsoft/azuredatastudio/pull/21590/files#diff-65d7c0b8cc76379cf757c90842b2a5cd45ca08e1a88411c44f71c46e2d8987b1R469